src/errorhandling/ErrorHandler.ts

```diff
@@ -85,16 +84,16 @@ export class ErrorHandler {
     // @ts-ignore
     const modules: MetadataModule[] = await apiPromise.runtimeMetadata.metadata
       .asV2.modules
-    let index = 0
     const filtered: MetadataModule[] = modules.filter((mod: MetadataModule) => {
       return !mod.events.isEmpty
     })
-    for (const m of filtered) {
+    let index = 0
+    filtered.forEach(m => {
       if (m.name.toString() === ErrorHandler.ERROR_MODULE_NAME) {
-        break
+        return
       }
-      index++
-    }
+      index += 1
+    })
 
```

Comment:
Not in the scope of this PR, but maybe this could be more compact/declarative: 
  
```javascript  
// find the last module error
return filtered.map(m => m.name.toString()).lastIndexOf(ErrorHandler.ERROR_MODULE_NAME); 
```   

(or on several lines if more readable)

Reply:
uhhh, nice